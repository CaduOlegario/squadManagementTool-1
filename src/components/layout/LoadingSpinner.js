import {withStyles} from "@material-ui/core";
import useTheme from "@material-ui/core/styles/useTheme";
import CircularProgress from "@material-ui/core/CircularProgress";
import PropTypes from 'prop-types';

const styles = {
    loading: {
        position: "fixed",
        width: "100%",
        height: "100%",
        top: "0",
        left: "0",
        right: "0",
        bottom: "0",
        backgroundColor: "rgba(0,0,0,0.5)",
        zIndex: "1100",
    },
    box: {
        padding: "54px 60vw",
        position: "absolute",
        top: "50%",
        left: "50%",
        transform: "translate(-50%,-50%)",
    },
    boxImg: {
        height: "150px",
        paddingBottom: "17px"
    },
    text: {
        textAlign: "center",
        fontSize: "16px",
        color: "#fff",
        display: "flex",
        alignItems: "center",
        marginLeft: "8px"
    },
    loader: {
        color: "white",
        marginRight: "5px"
    }
};

const LoadingSpinner = (props) => {
    const {visible, classes} = props;
    const theme = useTheme();

    return (
        visible && <div>
            <div className={classes.loading}>
                <div className={classes.box} style={{background: "linear-gradient(to right, " + theme.palette.primary.main + ", " + theme.palette.primary.dark + ")"}}>
                    <img className={classes.boxImg} src="/images/venturus_logo.png" alt="Venturus Logo" />
                    <br/>
                    <span className={classes.text}>
                        <CircularProgress className={classes.loader} size={22} />&nbsp;Carregando...
                    </span>
                </div>
            </div>
        </div>
    );
};

LoadingSpinner.propTypes = {
    visible: PropTypes.bool.isRequired,
    classes: PropTypes.object.isRequired
}

export default withStyles(styles)(LoadingSpinner);