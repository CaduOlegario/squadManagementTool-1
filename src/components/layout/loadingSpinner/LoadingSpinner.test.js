import React from 'react';
import renderer from 'react-test-renderer';
import LoadingSpinner from "./LoadingSpinner";

it('renders correctly normal input', () => {
    const tree = renderer
        .create(<LoadingSpinner visible={true}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
