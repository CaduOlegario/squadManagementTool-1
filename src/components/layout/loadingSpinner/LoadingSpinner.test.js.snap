// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly normal input 1`] = `
<div>
  <div
    className="LoadingSpinner-loading-1"
  >
    <div
      className="LoadingSpinner-box-2"
      style={
        Object {
          "background": "linear-gradient(to right, #3f51b5, #303f9f)",
        }
      }
    >
      <img
        alt="Venturus Logo"
        className="LoadingSpinner-boxImg-3"
        src="/images/venturus_logo.png"
      />
      <br />
      <span
        className="LoadingSpinner-text-4"
      >
        <div
          className="MuiCircularProgress-root LoadingSpinner-loader-5 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
          role="progressbar"
          style={
            Object {
              "height": 22,
              "width": 22,
            }
          }
        >
          <svg
            className="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
              cx={44}
              cy={44}
              fill="none"
              r={20.2}
              strokeWidth={3.6}
              style={Object {}}
            />
          </svg>
        </div>
        Â Carregando...
      </span>
    </div>
  </div>
</div>
`;
